### PR TITLE
sort object keys to our specification before generating user colormap

### LINKED ISSUE #1784: [BUG] Some Ident colors in key don't match data
**Describe the bug**
Some Ident colors in key don't always match the color on data. For some Ident groups, all the cluster colors match... and for some, only some of the clusters are wrong.

**To Reproduce**
1. Load
2. look at different ident groups

**Expected behavior**
Key color to match color on data

**Screenshots**
Here you see Cluster 4 of Ident group Integrated_snn_res.0.4 is orange in the key but green on the plot
image

**Version (please complete the following information):**
 - Desktop or hosted?: hosted (cellxgene-gateway)
 - Browser (if hosted) [e.g. chrome, safari]: chrome
 - Version [e.g. 0.13.0]: 0.16.2

**Additional context**

### PR DESCRIPTION
**Pull request by seve**
The order by which we were iterating over user defined colors was leading to a mismatch of color swathes on the left sidebar.  This PR reorders the keys before iteration to ensure that match the order by which we display and interact with them throughout the app.

---
Closes #1784

### CODE REVIEW

**Comment by colinmegill** [comment]
Great fix!

File: client/src/util/stateManager/colorHelpers.js [hunk 1]
```diff
@@ -94,15 +94,26 @@ export const createColorTable = memoize(_createColorTable);
...
-    const [colors, scaleMap] = Object.keys(userColors[category]).reduce(
-      (acc, label, i) => {
-        const color = parseRGB(userColors[category][label]);
-        acc[0][label] = color;
-        acc[1][i] = d3.rgb(255 * color[0], 255 * color[1], 255 * color[2]);
-        return acc;
-      },
-      [{}, {}]
-    );
+    // We cannot iterate over keys without sorting
```
Comment (tihuan) [inline client/src/util/stateManager/colorHelpers.js:97]: Thanks so much for the fix, @seve ! 🎉 Love the comment too!

Can you help write some unit tests to help catch these edge cases in the future too please? Thank you!
